### DISCUSSION TITLE
Some issues abourt LLVM JIT

### DISCUSSION THREAD

**Discussion by HungMingWu** [MikePopoloski/slang#585] [General]
Hi Mike, I met some issues about enable LLVM support.
1. Catch2 version incompatible  (I have fixed)
2. LLVM version is needs to version 10.0.X (Any plan to upgrade?)
3. The main issue is about function_ref
it crash on here.
``` cpp
slang::runtime::setOutputHandler([&](string_view text) { result += text; });
```
From the comment
``` cpp
// Work around a clang optimization bug on deallocating "result" before lambda call
[[clang::optnone]]
```
GCC have identical behavior,  So I change it to 
``` cpp
__attribute__((optimize("O0")))
```
Now it works. (My gcc is 11.2)
Whether change from function_ref to function a good idea?

**Comment by MikePopoloski** [comment]
Thanks for looking at this. The LLVM JIT code is highly experimential (and hasn't been built in a while, which is why you found it broken). Your changes seem fine to me, but at some point I need to look at the upgraded LLVM and see what changed. If you want to propose diffs to improve it that's fine, but again the design for codegen isn't stable or particularly well thought out right now.

As far as std::function vs function_ref goes, the idea is to get better performance by avoiding the std::function overhead if possible. Hopefully newer compiler versions fix their bug and then we won't have to worry about the optimization setting.
